src/main/java/seedu/meditracker/medication/MedicationList.java

```diff
@@ -0,0 +1,26 @@
+package seedu.meditracker.medication;
+
+import java.util.ArrayList;
+
+/**
+ * The MedicationList class represents a list of medications.
+ * It contains an ArrayList of Medication objects.
+ */
+public class MedicationList {
+    /** The list of medications stored in an ArrayList. */
+    public ArrayList<Medication> medications = new ArrayList<Medication>();
+
+    /**
+     * Constructs an empty MedicationList.
+     */
+    public MedicationList() {
+    }
+
+    /**
+     * Constructs a MedicationList with the specified list of medications.
+     * @param medications The list of medications to be stored.
+     */
+    public MedicationList(ArrayList<Medication> medications) {
```

Comment:
Might be better to use interface List instead of explicit type of list

```suggestion
    public MedicationList(List<Medication> medications) {
```